File: wallendar-app/app/src/main/java/com/td/wallendar/di/DependenciesContainer.java
package com.td.wallendar.di;

import android.content.Context;
import android.content.SharedPreferences;

import com.td.wallendar.repositories.interfaces.ChargesRepository;
import com.td.wallendar.repositories.interfaces.DebtsRepository;
import com.td.wallendar.repositories.interfaces.GroupsRepository;
import com.td.wallendar.service.ChargesService;
import com.td.wallendar.service.DebtsService;
import com.td.wallendar.service.GroupsService;
import com.td.wallendar.utils.login.LoginUtils;
import com.td.wallendar.utils.scheduler.SchedulerProvider;

import retrofit2.Retrofit;

public interface DependenciesContainer {
    Context getApplicationContext();

    SharedPreferences getLoginSharedPreferences();

    SchedulerProvider getSchedulerProvider();

    GroupsRepository getGroupsRepository();

    ChargesRepository getChargesRepository();

    DebtsRepository getDebtsRepository();

    Retrofit getRetrofit();

    GroupsService getGroupsService();

    ChargesService getChargesService();

    DebtsService getDebtsService();

    LoginUtils getLoginUtils();
}

File: wallendar-app/app/src/main/java/com/td/wallendar/di/DependenciesContainer.kt
package com.td.wallendar.di

import android.content.Context
import android.content.SharedPreferences
import com.td.wallendar.repositories.interfaces.*
import com.td.wallendar.service.*
import com.td.wallendar.utils.scheduler.SchedulerProvider
import retrofit2.Retrofit

interface DependenciesContainer {
    fun getApplicationContext(): Context
    fun getLoginSharedPreferences(): SharedPreferences
    fun getSchedulerProvider(): SchedulerProvider
    fun getGroupsRepository(): GroupsRepository
    fun getChargesRepository(): ChargesRepository
    fun getDebtsRepository(): DebtsRepository
    fun getApplicationUsersRepository(): ApplicationUsersRepository
    fun getRetrofit(): Retrofit
    fun getGroupsService(): GroupsService
    fun getChargesService(): ChargesService
    fun getDebtsService(): DebtsService
    fun getApplicationUsersService(): ApplicationUsersService
    fun getEventsRepository(): EventsRepository
    fun getEventsService(): EventsService
}